src/VisualStudio/Core/Def/Implementation/Workspace/Esent/EsentPersistentStorage.cs

```diff
@@ -274,12 +274,10 @@ private bool TryGetUniqueId(string value, bool fileCheck, out int id)
 
             try
             {
-                id = _nameTableCache.GetOrAdd(value, v =>
-                {
-                    // okay, get one from esent
-                    var uniqueIdValue = fileCheck ? FilePathUtilities.GetRelativePath(Path.GetDirectoryName(SolutionFilePath), v) : v;
-                    return _esentStorage.GetUniqueId(uniqueIdValue);
-                });
+                var uniqueIdValue = fileCheck ? FilePathUtilities.GetRelativePath(Path.GetDirectoryName(SolutionFilePath), value) : value;
+                id = _nameTableCache.GetOrAdd(value, _esentStorage.GetUniqueId(uniqueIdValue));
```

Comment:
And what are the time costs of the calls that were deferred before, but now always executed?  The trace says we allocate less, but are we trading that for a bunch of CPU time?

`FilePathUtilities.GetRelativePath(Path.GetDirectoryName(SolutionFilePath), value)` and `_esentStorage.GetUniqueId(uniqueIdValue)`